dev/com.ibm.ws.security.acme/src/com/ibm/ws/security/acme/internal/AcmeProviderImpl.java

```diff
@@ -290,7 +290,14 @@ public void revoke(List<X509Certificate> certificateChain, String reason) throws
 		acquireWriteLock();
 		try {
 			X509Certificate certificate = getLeafCertificate(certificateChain);
-			getAcmeClient().revoke(certificate, reason);
+			if (certificate == null) {
+				return;
+			}
+			String directoryURI = acmeHistory.getDirectoryURI(certificate.getSerialNumber().toString(16));
```

Comment:
Comment:

Check to see if the certificate is in the history file - it should be unless we are transitioning from self-signed to ACME. If the certificate isn't in the history file, use the configured directory URI.